File: lib/analyzers/explain.ts
type Label = "Safe" | "Suspicious" | "Phishing";

const BADGES: Record<Label, string> = {
  Safe: "آمن",
  Suspicious: "مريب",
  Phishing: "تصيّد",
};

export function buildExplanation(
  label: Label,
  confidence: number,
  reasons: string[]
): string {
  const badge = BADGES[label];

  const reasonsTxt =
    reasons.length > 0
      ? reasons.map((r) => `- ${r}`).join("\n")
      : "- لا توجد مؤشرات قوية، النتيجة مبنية على نمط عام.";

  return `${badge}
الثقة: ${Math.round(confidence * 100)}%

الأسباب:
${reasonsTxt}

نصيحة سريعة:
- لا تضغط أي رابط قبل التحقق من الدومين الحقيقي.
- لا تشارك OTP أو كلمة مرور عبر الرسائل.`;
}

File: app/services/explain.py
def build_explanation(label: str, confidence: float, reasons: list[str]) -> str:
    badge = {
        "Safe": "✅ آمن",
        "Suspicious": "⚠️ مريب",
        "Phishing": "🧨 تصيّد"
    }[label]

    reasons_txt = "\n".join([f"- {r}" for r in reasons]) if reasons else "- لا توجد مؤشرات قوية، النتيجة مبنية على نمط عام."
    return f"""{badge}
الثقة: {round(confidence*100, 1)}%

الأسباب:
{reasons_txt}

نصيحة سريعة:
- لا تضغط أي رابط قبل التحقق من الدومين الحقيقي.
- لا تشارك OTP أو كلمة مرور عبر الرسائل.
"""
